surfactant/filetypeid/id_extension.py

```diff
@@ -0,0 +1,38 @@
+# Copyright 2023 Lawrence Livermore National Security, LLC
+# See the top-level LICENSE file for details.
+#
+# SPDX-License-Identifier: MIT
+import pathlib
+import re
+from typing import Optional
+
+import surfactant.plugin
+
+
+@surfactant.plugin.hookimpl
+def identify_file_type(filepath: str) -> Optional[str]:
+    # pylint: disable=too-many-return-statements
+    try:
+        with open(filepath, "r") as f:
+            head = f.readline().strip("\n")
```

Comment:
I generated some SBOMs using the changes in this branch -- one issue I encountered is for binary files, Python will throw an exception (UnicodeDecodeError) if there are invalid utf-8 bytes. I think an initial check for just the "#!" starting a file (I suppose these are actually a human readable form of magic bytes) would address most of these cases, but we may want to look for the end of line `\n` byte by looking through the raw bytes first, before trying to decode them as a string.